packages/flutter/lib/src/widgets/drag_target.dart

```diff
@@ -418,26 +422,30 @@ class _DragAvatar<T> extends Drag {
     _entry.markNeedsBuild();
     HitTestResult result = new HitTestResult();
     WidgetFlutterBinding.instance.hitTest(result, globalPosition + feedbackOffset);
-    _DragTargetState<T> target = _getDragTarget(result.path);
-    if (target == _activeTarget)
+    List<_DragTargetState<T>> targets = _getDragTargets(result.path);
+    if (targets.contains(_activeTarget))
       return;
     if (_activeTarget != null)
       _activeTarget.didLeave(data);
-    _activeTarget = target;
+    _activeTarget = targets.firstWhere(
+      (_DragTargetState<T> target) => target.willAccept(data),
+      orElse: () => targets.isEmpty ? null : targets.first
+    );
     _activeTargetWillAcceptDrop = _activeTarget != null && _activeTarget.didEnter(data);
   }
 
-  _DragTargetState<T> _getDragTarget(List<HitTestEntry> path) {
-    // Look for the RenderBox that corresponds to the hit target (the hit target
-    // widget builds a RenderMetadata box for us for this purpose).
+  List<_DragTargetState<T>> _getDragTargets(List<HitTestEntry> path) {
```

Comment:
(this will then short-circuit the search as soon as you hit a match, instead of doing a deep walk each time)